### PR TITLE
[AMD][bugfix] Place TBO cuda-graph num_token_non_padded buffer on model devices

### PR DESCRIPTION
**Pull request by karverma-amd**
## Motivation
`TboCudaGraphRunnerPlugin` preallocates a persistent `_tbo_children_num_token_non_padded` buffer with a bare `torch.zeros((2,), dtype=torch.int32)`, which leaves it on **CPU**.
`ForwardBatch.num_token_non_padded` is contractually a tensor on the **model device** (`ForwardBatch.compute` does `.to(device, ...)`), and the eager TBO split path already honors this — `compute_tbo_children_num_token_non_padded_raw` moves its result to `get_global_server_args().device`. Only the cuda-graph plugin's preallocated buffer was left on CPU.
During TBO cuda-graph replay, that CPU buffer becomes the child's `num_token_non_padded` and is compared against 

## Modifications

- `python/sglang/srt/batch_overlap/two_batch_overlap.py`: allocate `_tbo_children_num_token_non_padded` on `get_global_server_args().device` so the cuda-graph buffer matches the eager split path and the `ForwardBatch` device contract. (`get_global_server_args` is already imported and used in this module.)
- `test/registered/unit/batch_overlap/test_tbo_cuda_graph_num_token_device.py`: new CPU-only regression test (registered via `register_cpu_ci`, `suite="base-a-test-cpu"`). It uses a `meta` device to make buffer placement observable without a GPU and asserts that (a) the plugin buffer lands on the model device, (b) the cuda-graph and eager paths agree on device, and (c) the eager split values are correct (`[min(split, n), max(0, n-split)]`).

## Accuracy Tests

export PYTHONPATH=/sgl-pr/python:${PYTHONPATH:-}
for r in 1 2 3; do
  python3 -m sglang.test.few_shot_gsm8k \
    --num-questions 1319 --num-shots 5 --parallel 32 \
    --host 127.0.0.1 --port 30000
done

image

## Speed Tests and Profiling

export PYTHONPATH=/sgl-pr/python:${PYTHONPATH:-}

# MoRI + aiter env
export SGLANG_AITER_MOE=1 TORCH_NCCL_BLOCKING_WAIT=1 TRITON_ENABLE_PRELOAD_KERNELS=0
export SGLANG_ROCM_FUSED_DECODE_MLA=1 SGLANG_AITER_AR=1
export ROCM_QUICK_REDUCE_QUANTIZATION=INT4 ROCM_QUICK_REDUCE_MAX_SIZE_BYTES_MB=512
export SGLANG_AITER_MLA_PERSIST=1 SGLANG_USE_AITER=1 TORCH_BLAS_PREFER_HIPBLASLT=1
export PYTORCH_ALLOC_CONF=expandable_segments:True
export MORI_SHMEM_MODE=ISOLATION MORI_SHMEM_HEAP_SIZE=16G
export SGLANG_MORI_DISPATCH_DTYPE=fp4 SGLANG_MORI_COMBINE_DTYPE=fp8
export SGLANG_MORI_NUM_MAX_DISPATCH_TOKENS_PER_RANK=2048

python3 -m sglang.launch_server \
  --model-path /models/DeepSeek-R1-0528-MXFP4 \
  --trust-remote-code \
  --tp-size 8 --ep-size 8 \
  --moe-a2a-backend mori --deepep-mode normal --moe-dense-tp-size 1 \
  --enable-dp-attention --dp-size 8 --enable-dp-lm-head \
  --enable-two-batch-overlap \
  --mem-fraction-static 0.55 \
  --max-running-requests 256 \
  --kv-cache-dtype fp8_e4m3 \
  --triton-attention-num-kv-splits 2 \
  --disable-radix-cache --disable-chunked-prefix-cache \
  --cuda-graph-bs 1 2 4 8 16 32 64 128 256 512 \
  --cuda-graph-max-bs 512 \
  --watchdog-timeout 1200 \
  --host 0.0.0.0 --port 30000

image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #27770502181](https://github.com/sgl-project/sglang/actions/runs/27770502181)
Latest PR Test (Extra): :x: [Run #27770501591](https://github.com/sgl-project/sglang/actions/runs/27770501591)

### CODE REVIEW

**Comment by billishyahao** [comment]
The change is trivial so I don't think we need to introduce a new unittest. And Could you run the e2e accuracy check and perf test as well? Thanks!

**Comment by karverma-amd** [comment]
> The change is trivial so I don't think we need to introduce a new unittest. And Could you run the e2e accuracy check and perf test as well? Thanks!

Done. Updated comment.

**Review by billishyahao — APPROVED** [review]
LGTM

**Review by HaiShaw — APPROVED** [review]

**Comment by HaiShaw** [comment]
@amd-bot ci-status
